include/seqan3/io/sequence/sequence_file_in.hpp

```diff
@@ -14,155 +58,123 @@ namespace seqan3
-    t::valid_formats;
-    requires detail::meets_concept_sequence_file_format<0, typename t::valid_formats>();
-
+    typename t::stream_type;
+    typename t::valid_format_types;
+    requires detail::meets_sequence_file_format_concept<typename t::valid_format_types>(
+        std::make_index_sequence<std::variant_size_v<typename t::valid_format_types>>{}
+    );
     t::valid_compression_formats;
 };
 
 struct sequence_file_in_default_traits
 {
-    using stream_type = std::ofstream;
-    using valid_formats = std::variant<sequence_file_format_fasta<stream_type>,
-                                       sequence_file_format_fastq<stream_type>,
-                                       sequence_file_format_embl<stream_type>,
-                                       sequence_file_format_genbank<stream_type>,
-                                       sequence_file_format_raw<stream_type>>;
-    static constexpr std::vector<std::pair<std::string, void>> valid_compression_formats{};
+    using stream_type = std::ifstream;
+    using valid_format_types = std::variant<sequence_file_format_fasta/*,
+                                       sequence_file_format_fastq,
+                                       sequence_file_format_embl,
+                                       sequence_file_format_genbank,
+                                       sequence_file_format_raw*/>;
+    using valid_compressions = std::variant<char/*specify compression formats*/>;
+    static inline std::vector<std::pair<std::string, valid_compressions>> valid_compression_formats{};
 };
 
 // ==================================================================
 // sequence_file_in
 // ==================================================================
 
-template <typename sequence_file_in_traits = sequence_file_in_traits_default>
+template <typename sequence_file_in_traits = sequence_file_in_default_traits>
     requires sequence_file_in_traits_concept<sequence_file_in_traits>
-class sequence_file_in
+class sequence_file_in : public detail::file_base<sequence_file_in_traits>
 {
 public:
-    using sequence_file_in_traits::stream_type;               // e.g. std::ostream concept
-    using sequence_file_in_traits::valid_compression_formats; // = std::vector<std::pair<std::string, t>>
-    using sequence_file_in_traits::valid_formats;             // = std::variant<gzip_compressor, sequence_file_format_bam>
-
-    // constructor with arg
-    sequence_file_in(std::string const & _file_name);
-
-    // copy construction and assignment are deleted
-    // implicitly because we don't want multiple access to file
+    /* types */
+    using stream_type = typename detail::file_base<sequence_file_in_traits>::stream_type;
+    using valid_format_types = typename detail::file_base<sequence_file_in_traits>::valid_format_types;
+
+    /* constructors */
+    sequence_file_in(std::experimental::filesystem::path _file_name) :
+        detail::file_base<sequence_file_in_traits>(_file_name) {};
```

Comment:
move the _file_name in the constructor call of the base class.
declare the constructor ```explicit```.